rs/tests/networking/canister_http_correctness_test.rs

```diff
@@ -1661,44 +1660,6 @@ fn test_only_headers_with_custom_max_response_bytes_exceeded(env: TestEnv) {
     );
 }
 
-fn test_non_ascii_url_is_rejected(env: TestEnv) {
```

Comment:
This should not be dropped, but updated to check that everything works with non-ASCII characters in the URL.

Reply:
E.g., hyper [panics](https://github.com/dfinity/ic/blob/0e5bc0d1a18b287eb70f5b2b8a30099c91e9de4c/rs/https_outcalls/adapter/src/rpc_server.rs#L37) if there are too many request headers so we should make sure that non-ASCII urls do not result in a panic either

Reply:
Ok, added `test_non_ascii_url_is_accepted` to check that